src/inmanta/resources.py

```diff
@@ -427,7 +427,10 @@ def map_field(
             elif hasattr(cls, "map") and field_name in cls.map:
                 value = cls.map[field_name](exporter, model_object)
             else:
-                value = getattr(model_object, field_name)
+                try:
+                    value = getattr(model_object, field_name)
+                except inmanta.ast.UndeclaredReference as e:
+                    value = e.reference
```

Comment:
Why is this? I.e. why would this work/be correct when we abort at the first reference? The reference can be part of the attribute, it doesn't have to be the entire attribute

Moreover, why would we get this exception here? References are expected in resource serialization. It is more or less the whole point of them. We should not run this in the new mode we use for plugins.

Reply:
Yeah, this module needs some more work, I'm aware of it. I'll make sure to take this comment into consideration.

> Moreover, why would we get this exception here? References are expected in resource serialization. It is more or less the whole point of them. We should not run this in the new mode we use for plugins.

That's sort of the point of this structure here. Though I think I may have made this change before I added the `allow_references()` method on proxy.